File: Leetcode/371. Sum of Two Integers/sum_of_two_integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xffffffff
        while b !=0:
            carry = (a&b)<<1
            a = (a^b)& mask
            b = carry & mask
        if a > mask // 2:
            return ~(a ^ mask)
        else:
            return a
sol = Solution()
print(sol.getSum(-12,-8))

File: Leetcode/371. Sum of Two Integers/sum_of_two_integers.rs
struct Solution{}

impl Solution {
    pub fn get_sum(mut a: i32, mut b: i32) -> i32 {
        while b!=0{
            let carry = (a&b)<<1;
            a = a^b;
            b = carry;
        }
        return a
    }
}
fn main() {
    let res = Solution::get_sum(-12,-90);
    println!("{:?}",res);
}
